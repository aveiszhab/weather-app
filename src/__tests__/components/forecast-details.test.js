import React from 'react';
import { render } from '@testing-library/react';
import ForecastDetails from '../../components/forecast-details';

const mockForecast = {
    date: 1525046400000,
    temperature: { max: 11, min: 4 },
    humidity: 30,
    wind: { speed: 13, direction: "s" }
}

describe('ForecastDetails', () => {
    it('renders correctly to match ForecastDetails snapshot', () =>{
        const { asFragment } = render(
            < ForecastDetails forecast = { mockForecast } />
        );

        expect(asFragment()).toMatchSnapshot()
    });

    it('renders the correct date, temperature, humidity and wind props', () => {
        const { getByText} = render(
            <ForecastDetails forecast = { mockForecast } />
        );

        expect(getByText('Mon 30th Apr')).toHaveClass('date');
        expect(getByText('Max Temperature: 11°c')).toHaveClass('temperatureMax');
        expect(getByText('Min Temperature: 4°c')).toHaveClass('temperatureMin');
        expect(getByText('Humidity: 30%')).toHaveClass('humidity');
        expect(getByText('Wind: 13mph s')).toHaveClass('wind');

    });
    
});