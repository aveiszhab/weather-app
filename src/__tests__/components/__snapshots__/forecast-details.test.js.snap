// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly to match ForecastDetails snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="date"
    >
      Mon 30th Apr
    </div>
    <div
      class="temperatureMax"
    >
      Max Temperature: 11°c
    </div>
    <div
      class="temperatureMin"
    >
      Min Temperature: 4°c
    </div>
    <div
      class="humidity"
    >
      Humidity: 30%
    </div>
    <div
      class="wind"
    >
      Wind: 13mph s
    </div>
  </div>
</DocumentFragment>
`;
